hoot-core/src/main/cpp/hoot/core/io/OsmApiWriter.cpp

```diff
@@ -225,7 +225,8 @@ void OsmApiWriter::_changesetThreadFunc()
       //  Display the changeset in TRACE mode
       LOG_TRACE("Thread: " << std::this_thread::get_id() << "\n" << _changeset.getChangesetString(workInfo, id));
       //  Upload the changeset
-      if (_uploadChangeset(request, id, _changeset.getChangesetString(workInfo, id)))
+      OsmApiFailureInfoPtr info = _uploadChangeset(request, id, _changeset.getChangesetString(workInfo, id));
```

Comment:
`_uploadChangeset` now returns an object pointer that contains more information, including error message, if the changeset fails to upload.  This allows the offending element to be culled from the changeset more efficiently instead of bisecting the changeset.